### PR TITLE
feat: allowed callers interface

### PR DESCRIPTION
**Pull request by davidprepo**

### CODE REVIEW

File: apps/smart-contracts/core/contracts/interfaces/IAllowedCallers.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   */
+  event AllowedCallersChange(address[] callers, bool[] allowed);
+
+  /**
+   * @notice Sets an array of addresses to be allowed or not.
+   * @dev The length of `callers` and `allowed` must match.
+   * @param callers Addresses to change inclusion for
+   * @param allowed Whether to allow corresponding address
+   */
+  function setAllowedCallers(address[] memory callers, bool[] memory allowed)
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/contracts/interfaces/IAllowedCallers.sol:19]:
```suggestion
  function setAllowedCallers(address[] memory _callers, bool[] memory _allowed)
```

File: apps/smart-contracts/core/contracts/interfaces/IAllowedCallers.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    external;
+
+  /**
+   * @notice Getter for whether an address is allowed to call
+   */
+  function isCallerAllowed(address caller) external view returns (bool);
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/contracts/interfaces/IAllowedCallers.sol:25]:
```suggestion
  function isCallerAllowed(address _caller) external view returns (bool);
```

**Review by ramenforbreakfast — APPROVED** [review]
LGTM, besides my comments regarding parameters

**Review by XavierEkkel — APPROVED** [review]
LGTM, noting that we will fix up the natspec later (in future, please leave natspec until after implementation)

**Pull request merged**